palette/src/rgb/packed.rs

```diff
@@ -0,0 +1,364 @@
+///
+/// See [PackedRgb](struct.PackedRgb.html) for more details.
+pub struct RgbaPacked;
+impl RgbChannels for RgbaPacked {
+    fn split_rgb<S: RgbStandard>(rgb: Rgba<S, u8>) -> (u8, u8, u8, u8) {
+        (rgb.red, rgb.green, rgb.blue, rgb.alpha)
+    }
+
+    fn combine_rgb<S: RgbStandard>(channels: (u8, u8, u8, u8)) -> Rgba<S, u8> {
+        Rgba::new(channels.0, channels.1, channels.2, channels.3)
+    }
+}
+
+/// RGBA color packed in ARGB order.
+///
+/// See [PackedRgb](struct.PackedRgb.html) for more details.
+pub struct Argb;
+impl RgbChannels for Argb {
+    fn split_rgb<S: RgbStandard>(rgb: Rgba<S, u8>) -> (u8, u8, u8, u8) {
+        (rgb.alpha, rgb.red, rgb.green, rgb.blue)
+    }
+
+    fn combine_rgb<S: RgbStandard>(channels: (u8, u8, u8, u8)) -> Rgba<S, u8> {
+        Rgba::new(channels.1, channels.2, channels.3, channels.0)
+    }
+}
+
+/// RGBA color packed in BGRA order.
+///
+/// See [PackedRgb](struct.PackedRgb.html) for more details.
+pub struct Bgra;
+impl RgbChannels for Bgra {
+    fn split_rgb<S: RgbStandard>(rgb: Rgba<S, u8>) -> (u8, u8, u8, u8) {
+        (rgb.blue, rgb.green, rgb.red, rgb.alpha)
+    }
+
+    fn combine_rgb<S: RgbStandard>(channels: (u8, u8, u8, u8)) -> Rgba<S, u8> {
+        Rgba::new(channels.2, channels.1, channels.0, channels.3)
+    }
+}
+
+/// RGBA color packed in ABGR order.
+///
+/// See [PackedRgb](struct.PackedRgb.html) for more details.
+pub struct Abgr;
+impl RgbChannels for Abgr {
+    fn split_rgb<S: RgbStandard>(rgb: Rgba<S, u8>) -> (u8, u8, u8, u8) {
+        (rgb.alpha, rgb.blue, rgb.green, rgb.red)
+    }
+
+    fn combine_rgb<S: RgbStandard>(channels: (u8, u8, u8, u8)) -> Rgba<S, u8> {
+        Rgba::new(channels.3, channels.2, channels.1, channels.0)
+    }
+}
+
+impl<C: RgbChannels> From<u32> for PackedRgb<C> {
+    fn from(color: u32) -> Self {
+        PackedRgb {
+            color,
+            channel_order: PhantomData,
+        }
+    }
+}
+
+impl<C: RgbChannels> Into<u32> for PackedRgb<C> {
+    fn into(self) -> u32 {
+        self.color
+    }
+}
+
+impl<S, C> Into<Rgb<S, u8>> for PackedRgb<C>
+where
+    S: RgbStandard,
+    C: RgbChannels,
+{
+    fn into(self) -> Rgb<S, u8> {
+        let bytes: [u8; 4] = self.color.to_be_bytes();
+        let color: Alpha<Rgb<S, u8>, u8> = C::combine_rgb((bytes[0], bytes[1], bytes[2], bytes[3]));
+        Rgb::from_components((color.red, color.green, color.blue))
+    }
+}
+
+impl<S, C> From<Rgb<S, u8>> for PackedRgb<C>
+where
+    S: RgbStandard,
+    C: RgbChannels,
+{
+    fn from(color: Rgb<S, u8>) -> Self {
+        let src: Alpha<Rgb<S, u8>, u8> =
+            Rgba::from_components((color.red, color.green, color.blue, 0u8));
+        let bytes = C::split_rgb(src);
+        PackedRgb {
+            color: 0x0000_0000
+                | ((bytes.0 as u32) << 24)
+                | ((bytes.1 as u32) << 16)
+                | ((bytes.2 as u32) << 8)
+                | (bytes.3 as u32),
+            channel_order: PhantomData,
```

Comment:
This could let the `From<Rgba>` implementation do the bit shifting. :slightly_smiling_face: It could just end with `Self::from(src)`. The same goes for `Into`.